### ISSUE TITLE
weight file which made by train.py was not worked

### ISSUE THREAD

**Issue by wonny2001** [hizhangp/yolo_tensorflow#24]
in config.py
i modified just

WEIGHTS_FILE = None
BATCH_SIZE = 5

BUT!!
after training abt 3hours,
i did test.py, but there are no box on imname = 'test/cat.jpg'
..
what is the problem?? I just want to get my own weights file by my training

when i train with WEIGHTS_FILE = os.path.join(DATA_PATH, 'weights', 'YOLO_small.ckpt')
it worked well

**Comment by hizhangp** [comment]
You should finetune from the pretrained model.

**Comment by wonny2001** [comment]
how to finetune ?
this codes doesn't have a finetue? just for test??

**Comment by hizhangp** [comment]
Please refer to `train.py`, it gives you an example about how to finetune a model from pretrained model.

**Comment by wonny2001** [comment]
You mean like code below?

135 def main(): 
136     parser = argparse.ArgumentParser() 
137     parser.add_argument('--weights', default="YOLO_small.ckpt", type=str) 
138     parser.add_argument('--data_dir', default="data", type=str) 
139     parser.add_argument('--threshold', default=0.2, type=float) 
140     parser.add_argument('--iou_threshold', default=0.5, type=float) 
141     parser.add_argument('--gpu', default='', type=str) 


The thing is, 
Why i try with only pretrained model??

I want to change or add more "slim.conv2d" on below codes,
(ex. net = slim.conv2d(net, 256, 1, scope='conv_11_2')) 
To do this, i need to train from "WEIGHTS_FILE = None" unless it gives me error (bcz pretrained model is not same to mode which i changed)

55                 net = tf.pad(images, np.array([[0, 0], [3, 3], [3, 3], [0, 0]]), name='pad_1') 
56                 net = slim.conv2d(net, 64, 7, 2, padding='VALID', scope='conv_2') 
57                 net = slim.max_pool2d(net, 2, padding='SAME', scope='pool_3') 
58                 net = slim.conv2d(net, 192, 3, scope='conv_4') 
59                 net = slim.max_pool2d(net, 2, padding='SAME', scope='pool_5') 
60                 net = slim.conv2d(net, 128, 1, scope='conv_6') 
61                 net = slim.conv2d(net, 256, 3, scope='conv_7') 
62                 net = slim.conv2d(net, 256, 1, scope='conv_8') 
63                 net = slim.conv2d(net, 512, 3, scope='conv_9') 
64                 net = slim.max_pool2d(net, 2, padding='SAME', scope='pool_10') 
65                 net = slim.conv2d(net, 256, 1, scope='conv_11') 
66                 net = slim.conv2d(net, 512, 3, scope='conv_12') 
67                 net = slim.conv2d(net, 256, 1, scope='conv_13') 
68                 net = slim.conv2d(net, 512, 3, scope='conv_14') 
69                 net = slim.conv2d(net, 256, 1, scope='conv_15') 
70                 net = slim.conv2d(net, 512, 3, scope='conv_16') 
71                 net = slim.conv2d(net, 256, 1, scope='conv_17') 
72                 net = slim.conv2d(net, 512, 3, scope='conv_18') 
73                 net = slim.conv2d(net, 512, 1, scope='conv_19') 
74                 net = slim.conv2d(net, 1024, 3, scope='conv_20') 
75                 net = slim.max_pool2d(net, 2, padding='SAME', scope='pool_21') 
76                 net = slim.conv2d(net, 512, 1, scope='conv_22') 
77                 net = slim.conv2d(net, 1024, 3, scope='conv_23') 
78                 net = slim.conv2d(net, 512, 1, scope='conv_24') 
79                 net = slim.conv2d(net, 1024, 3, scope='conv_25') 
80                 net = slim.conv2d(net, 1024, 3, scope='conv_26') 
81                 net = tf.pad(net, np.array([[0, 0], [1, 1], [1, 1], [0, 0]]), name='pad_27') 
82                 net = slim.conv2d(net, 1024, 3, 2, padding='VALID', scope='conv_28') 
83                 net = slim.conv2d(net, 1024, 3, scope='conv_29') 
84                 net = slim.conv2d(net, 1024, 3, scope='conv_30') 
85                 net = tf.transpose(net, [0, 3, 1, 2], name='trans_31') 
86                 net = slim.flatten(net, scope='flat_32') 
87                 net = slim.fully_connected(net, 512, scope='fc_33') 
88                 net = slim.fully_connected(net, 4096, scope='fc_34') 
89                 net = slim.dropout(net, keep_prob=keep_prob, 
90                                    is_training=is_training, scope='dropout_35') 
91                 net = slim.fully_connected(net, num_outputs, 
92                                            activation_fn=None, scope='fc_36')

**Comment by hizhangp** [comment]
If you add extra layers, you should avoid loading these layers' weight from pretrainded model.

**Comment by feitiandemiaomi** [comment]
@wonny2001  Hi , I met the smilar question , How do you solve it ?
